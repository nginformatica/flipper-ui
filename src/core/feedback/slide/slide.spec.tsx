import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Slide } from '.'

describe('Slide', () => {
    it('should render closed', () => {
        render(
            <Slide direction='left' in={false}>
                <span>Slide Text</span>
            </Slide>
        )
        const element = screen.getByText('Slide Text')

        expect(element).toHaveProperty('style.visibility', 'hidden')
    })

    it('should render opened', () => {
        render(
            <Slide in direction='left'>
                <span>Slide Text</span>
            </Slide>
        )

        const element = screen.getByText('Slide Text')

        expect(element).not.toHaveProperty('style.visibility', 'hidden')
    })

    it('should match snapshot', () => {
        const { container: opened } = render(
            <Slide in direction='left'>
                <span>Slide Text</span>
            </Slide>
        )

        const { container: closed } = render(
            <Slide direction='left' in={false}>
                <span>Slide Text</span>
            </Slide>
        )

        expect(opened).toMatchSnapshot()
        expect(closed).toMatchSnapshot()
    })
})
