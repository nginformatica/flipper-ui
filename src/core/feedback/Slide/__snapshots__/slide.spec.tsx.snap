// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should match snapshot 1`] = `
<div>
  <span
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    Slide Text
  </span>
</div>
`;

exports[`Slide should match snapshot 2`] = `
<div>
  <span
    style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
  >
    Slide Text
  </span>
</div>
`;
